internal/datasource/git.go

```diff
@@ -280,6 +270,112 @@ func (s *GitSource) Get(
 	}, closer, nil
 }
 
+func (s *GitSource) Changes(
+	ctx context.Context,
+	log hclog.Logger,
+	ui terminal.UI,
+	raw *pb.Job_DataSource,
+	current *pb.Job_DataSource_Ref,
+) (*pb.Job_DataSource_Ref, error) {
+	source := raw.Source.(*pb.Job_DataSource_Git)
+
+	// Build our auth mechanism
+	auth, err := s.auth(log, ui, source)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get our remote
+	remote := git.NewRemote(memory.NewStorage(), &config.RemoteConfig{
+		Name: "origin",
+		URLs: []string{source.Git.Url},
+	})
+
+	// List our refs, equivalent to git ls-remote
+	refs, err := remote.List(&git.ListOptions{Auth: auth})
+	if err != nil {
+		return nil, err
+	}
+
+	// Find the target for HEAD
```

Comment:
Actually, this might be totally broken for some cases because we're just finding HEAD for the default branch when we probably need to look for the latest commit of the given ref for the project. Whoops!

Reply:
I'm going to propose fixing this in a follow-up PR (before 0.3.0) if that's okay. I'll follow on right after this.